pkg/urlutil/urlutil_test.go

```diff
@@ -6,17 +6,21 @@ var (
 	gitUrls = []string{
 		"git://github.com/docker/docker",
 		"git@github.com:docker/docker.git",
-		"git@bitbucket.org:atlassianlabs/atlassian-docker.git",
+		"git@bitbucket.org:atlassianlabs/atlassian-docker",
 		"https://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git#branch",
 		"http://github.com/docker/docker.git#:dir",
+		"git+ssh://github.com/docker/docker",
+		"git+https://github.com/docker/docker.git#branch",
+		"git+ftps://github.com/docker/docker.git#:dir",
```

Comment:
I think we should pick an explicit list of protocols we support - I suspect the current code will happily accept `git+mail://foo.bar`.

It's better to _restrict_ options, and (if desirable) later extend the list of accepted options, than to _allow_ all options, and later having to deprecate support for one (or more) of them